create-stack/main_test.go

```diff
@@ -0,0 +1,257 @@
+package main_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+	assertpkg "github.com/stretchr/testify/assert"
+	requirepkg "github.com/stretchr/testify/require"
+)
+
+const (
+	DistroName    = "Ubuntu"
+	DistroVersion = "18.04"
+	Homepage      = "https://github.com/paketo-buildpacks/stacks"
+	BionicStackID = "io.buildpacks.stacks.bionic"
+	Maintainer    = "Paketo Buildpacks"
+)
+
+func TestEntrypoint(t *testing.T) {
+	spec.Run(t, "Entrypoint", testEntrypoint, spec.Report(report.Terminal{}))
+}
+
+func testEntrypoint(t *testing.T, when spec.G, it spec.S) {
+	var (
+		cliPath string
+		require = requirepkg.New(t)
+		assert  = assertpkg.New(t)
+	)
+
+	it.Before(func() {
+		tempFile, err := ioutil.TempFile("", "entrypoint")
+		require.NoError(err)
+
+		cliPath = tempFile.Name()
+		require.NoError(tempFile.Close())
+
+		goBuild := exec.Command("go", "build", "-o", cliPath, ".")
+		output, err := goBuild.CombinedOutput()
+		require.NoError(err, "failed to build CLI: %s", string(output))
+	})
+
+	it.After(func() {
+		_ = os.Remove(cliPath)
+	})
+
+	it("builds base bionic-stack", func() {
+		buildRepo := "paketobuildpacks/test-create-stack-base-build"
+		runRepo := "paketobuildpacks/test-create-stack-base-run"
```

Comment:
Do we want to push test images here? It looks like there are no other test images there other than the stack-rcs. There are builder-rcs but it looks like they've stopped putting new ones there so they must have a different way of testing.

Are there any alternatives? Or could we test this without pushing the images anywhere?

Reply:
I do think we should test the publish case because there is some logic around setting the `base-image` metadata properly. But I agree that maybe there's a better place to push the test images since there are no other test/RC images in `paketobuildpacks`.

Reply:
There is a now [paketotesting](https://hub.docker.com/u/paketotesting) we can use for test images

Reply:
We've switched this to `paketotesting` now.